datafusion/sql/src/expr/function.rs

```diff
@@ -90,6 +90,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             let partition_by = window
                 .partition_by
                 .into_iter()
+                //ignore window spec PARTITION BY for scalar values
```

Comment:
```suggestion
                // ignore window spec PARTITION BY for scalar values
                // as they do not change and thus do not generate new partitions
```

I also double checked https://docs.rs/sqlparser/latest/sqlparser/ast/enum.Value.html is indeed the right node for literals